### PR TITLE
 Add Zookeeper upgrade test and integrate integration tests into Frameworks CI.

### PR DESCRIPTION
**Pull request by jbarrick-mesosphere**
We'll run tests against a KIND cluster in Circle CI for now.

I'll write user documentation for this in a follow up PR for kudobuilder/kudo#340.

We'll drop the Go stuff and integrate it into kudoctl in kudobuilder/kudo#352.

Fixes kudobuilder/kudo#353

### CODE REVIEW

**Comment by jbarrick-mesosphere** [comment]
For some reason, the Zookeeper Framework test fails upgrading the CPU requirements:

* https://circleci.com/gh/jbarrick-mesosphere/frameworks/18

This seems to be an issue with KUDO, as far as I can tell, so I will [fix the test to not make any changes to the statefulset](https://github.com/kudobuilder/frameworks/pull/23/commits/a2b8d6182ac30f92713486a19e45386016b97859) and then file a ticket about this with instructions to reproduce it once this is merged.

**Comment by runyontr** [comment]
@jbarrick-mesosphere No need to update a ticket:  https://github.com/kudobuilder/kudo/issues/58 It's been a known issue

File: frameworks_test.go
Comment (zmalik) [inline frameworks_test.go:18]: can we add `goimports` check from the start? 

```
 -		T:         t,
 -		TestDirs:  []string{
 +		T: t,
 +		TestDirs: []string{
  			"./repository/zookeeper/tests/",
  		},
```

`goimports` would fail on this line, just detected this in CI builds where we run `goimports`

File: Makefile
Comment (alenkacz) [inline Makefile:33]: this won't work after we release 0.3.0. You will have to use CLI to install.

We could either postpone this until 0.3.0 and change it or we need to fix it right after merging so it does not block the release

File: frameworks_test.go
Comment (jbarrick-mesosphere) [reply frameworks_test.go:18]: I'd prefer not to spend time on that, as this isn't a Go repository, we'll be dropping this in favor of running the harness with kudoctl: kudobuilder/kudo#352

File: Makefile
Comment (jbarrick-mesosphere) [reply Makefile:33]: Got it - this is the bare minimum needed to get the tests running on CI, I'll be iterating from here.

File: frameworks_test.go
Comment (runyontr) [inline frameworks_test.go]: I was picturing the tests unique for each version of a framework, not at the top level.  

./repository/zookeeper/0.1.0/tests <- Tests for FrameworkVersion 0.1.0
./repository/zookeeper/0.2.0/tests < - Tests for FV 0.2.0

File: frameworks_test.go
Comment (runyontr) [inline frameworks_test.go]: Could we just look for all test folders instead of having these hardcoded?

File: frameworks_test.go
Comment (oliviabarrick) [reply frameworks_test.go]: The repository format KEP says its `repository/zookeeper/tests/` (all versions) and `repository/zookeeper/tests/0.1.0`. I think I agree with you, but I’m still tossing around how I want that to work, so let’s keep it how we have it for now and I’ll come through and update it once we’ve figured it out on the KEP?

Comment (runyontr) [reply frameworks_test.go]: SGTM

**Review by runyontr — APPROVED** [review]

File: frameworks_test.go
Comment (jbarrick-mesosphere) [reply frameworks_test.go]: Changed to a glob for now. Will merge once tests pass.

**Pull request merged**
